storybook/sidebar.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line @typescript-eslint/no-restricted-imports
+import { createElement, useMemo } from 'react';
+import { useStorybookApi } from '@storybook/manager-api';
+import { styled } from '@storybook/theming';
+
+/**
+ * Internal dependencies
+ */
+import badges from './badges';
+
+const STEM = 'sb-';
+
+const Wrapper = styled.span( {
+	flexGrow: 1,
+	display: 'flex',
+	paddingRight: '20px',
+} );
+
+const Title = styled.span( {
+	flexGrow: 1,
+} );
+
+const Icons = styled.span( {} );
+
+const Icon = styled.span( {} );
+
+function useIcons( item ) {
+	const api = useStorybookApi();
+
+	return useMemo( () => {
+		let data = {};
+
+		if ( item.isComponent && item.children?.length ) {
+			data = api.getData( item.children[ 0 ] ) ?? {};
+		}
+
+		const { tags = [] } = data;
+
+		return tags
+			.filter( ( tag ) => tag.startsWith( STEM ) )
+			.map( ( tag ) => badges[ tag.substring( STEM.length ) ] )
+			.map( ( { icon, title, tooltip } ) =>
+				icon
+					? createElement(
+							Icon,
+							{ title: tooltip?.title ?? title },
+							icon
+					  )
+					: null
+			);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ item.id ] );
```

Comment:
We've generally found that following the `exhaustive-deps` linting is helpful for preventing bugs, even if some of the deps could _technically_ be unnecessary to list. If there is a super clear reason to override it though, let's add a code comment so the intention is clear.

Reply:
Resolved in 874b85e46a7507583836c6929bb72ccae64acd0a